WSL2: Ensure the Windows Terminal is detected under WSL2 docker build.

xt3firm opened the pull request:
- by passing the WT_SESSION environment variable to the docker container

xt3firm commented:
env variables SSH_CLIENT and PESTER_TERMINAl might be also candidates for passing due to the same reason

rpardini commented:
Interesting. I had no reports of Docker-under-WSL2 working before, and even had https://github.com/armbian/build/issues/4855 to block it. How did you achieve this...?

xt3firm commented:
What exactly is the question? I used WSL2 builds all over the time. Sometimes  it fails with rather strange errors, most of them because downloading failed, sometimes at the end in the cleaning phase. Besides of that, worked, and I am not sure if these failures are wsl2 problems. Typically restarting the build fixes it. Good enough fir me. Unfortunately I  used the master branch for long time, switched only yesterday to main, so I cannot say a lot for this branch,  however build a rock5 image without problems yesterday. Used wsl2 "productive" at work, has its weaknesses,  but after I switched to the current version with systemd support,  I'm really happy, not worser than colinux in the good ol' times. Am 08.03.23, 18:15 schrieb Ricardo Pardini ***@***.***>:

   
  Interesting. I had no reports of Docker-under-WSL2 working before, and even had #4855 to block it. How did you achieve this...? 
  —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

xt3firm commented:
BTW: I do not use docker desktop,  but the "real docker-ce" under ms debian 11 wsl2 image. Not sure if I did something "extra", believe it just runs with its default systemd support.--Diese Nachricht wurde von meinem Android Mobiltelefon mit GMX Mail gesendet.Am 08.03.23, 18:15 schrieb Ricardo Pardini ***@***.***>:

   
  Interesting. I had no reports of Docker-under-WSL2 working before, and even had #4855 to block it. How did you achieve this...? 
  —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

rpardini commented:
Oh -- the differences between Docker CE (native Windows binaries) and `docker.io` Ubuntu package under WSL2 are substantial. So I will rephrase #4855 to be clear it's only about Docker CE.

rpardini reviewed (changes requested):
Let's only add the `--env` if `WT_SESSION` actually is set, otherwise other OS's will have an empty `WT_SESSION` for no reason...

xt3firm commented:
Updated. BTW: I use the Linux docker binaries, just not the debian ones (if there are any).

$ docker --version
Docker version 20.10.5+dfsg1, build 55c4c88
